src/os/bluestore/BlueStore.cc

```diff
@@ -5432,7 +5432,7 @@ int BlueStore::_balance_bluefs_freespace(PExtentVector *extents)
   uint64_t min_free = cct->_conf.get_val<Option::size_t>("bluestore_bluefs_min_free");
   if (bluefs_free < min_free &&
       min_free < free_cap) {
-    uint64_t g = min_free - bluefs_free;
+    uint64_t g = std::max<uint64_t>(min_free - bluefs_free, cct->_conf->bluestore_bluefs_gift_ratio * total_free);
```

Comment:
@ligangbin117 IIUC, the description of "bluestore_bluefs_gift_ratio" reads

> Maximum fraction of free space to give to BlueFS at once

but you are using it as a minimum fraction here? could you elaborate the rationale a little bit?

Reply:
lots of like 34M,66M space allocated when  the bluefs free space is smaller than min_free.
Why not allocate more at once?
own 0x[885600000~2200000,888600000~4200000,88d300000~4100000,891600000~100000,afd000000~1a00000,aff100000~2800000,b02b00000~4200000,b07800000~3500000,c00000000~100000000,e75100000~c00000,e76f00000~4200000,e7bf00000~4200000,e80900000~2800000,10ed500000~1a00000,10efb00000~4200000,10f4400000~1b00000,1360a00000~2800000,1363f00000~4200000,1368d00000~4200000,136dd00000~4200000,1372d00000~4200000,1377a00000~4200000,137bf00000~d00000,15e7500000~2600000]

Reply:
perhaps we should update the description of the gift_ratio option to remove the 'Maximum' part

Reply:
IMO this will violate 'free_cap' constraint which might be a bit confusing,